src/BaseItemManyWidgets.h

```diff
@@ -0,0 +1,149 @@
+// Creator: @ShishkinDmitriy
+#ifndef BASE_ITEM_MANY_WIDGETS_H
+#define BASE_ITEM_MANY_WIDGETS_H
+
+#include "LcdMenu.h"
+#include "MenuItem.h"
+#include "widget/Widget.h"
+#include <utils/utils.h>
+
+class BaseItemManyWidgets : public MenuItem {
+  protected:
+    BaseWidget** widgets = nullptr;
+    const uint8_t size = 0;
+    uint8_t activeWidget = 0;
+
+  public:
+    BaseItemManyWidgets(const char* text, BaseWidget** widgets, const uint8_t size, uint8_t activeWidget = 0)
+        : MenuItem(text), widgets(widgets), size(size), activeWidget(activeWidget) {}
+
+    uint8_t getActiveWidget() const { return activeWidget; }
+    void setCurrent(const uint8_t newCurrent) { activeWidget = newCurrent; }
+
+  protected:
+    virtual void handleCommit() = 0;
+    /**
+     * @brief Reset the active widget to the first widget.
+     */
+    void reset() { activeWidget = 0; }
+
+    /**
+     * @brief Draws the menu item using the provided renderer.
+     *
+     * This function is responsible for rendering the menu item and its associated widgets.
+     * It iterates through each widget, drawing them into a buffer, and then uses the renderer
+     * to display the combined result. If the renderer is in edit mode, it also handles the
+     * cursor positioning for editing.
+     *
+     * @param renderer A pointer to the MenuRenderer object used for drawing the item.
+     *
+     * The function performs the following steps:
+     * 1. Initializes a buffer to hold the drawn content and variables for indexing and cursor positioning.
+     * 2. Iterates through each widget, drawing it into the buffer and updating the index.
+     * 3. If the active widget is being edited, it draws the item using the renderer and calculates the cursor position.
+     * 4. Draws the final item using the renderer.
+     * 5. If in edit mode, moves the cursor to the appropriate position for editing.
+     */
+    void draw(MenuRenderer* renderer) override {
+        char buf[ITEM_DRAW_BUFFER_SIZE];
+
+        uint8_t index = 0;
+        uint8_t cursorCol = 0;
+
+        for (uint8_t i = 0; i < size; i++) {
+            index += widgets[i]->draw(buf, index);
+            if (i == activeWidget && renderer->isInEditMode()) {
+                renderer->drawItem(text, buf);
+                cursorCol = renderer->getCursorCol() - 1 - widgets[i]->blinkerOffset;
+            }
+        }
+        renderer->drawItem(text, buf);
+
```

Comment:
@coderabbitai unfortunately, it is required to draw a widget in order to get the cursor position, that's why I draw it so that later on when it is in edit mode, the blinker will be correctly placed.